### DISCUSSION TITLE
Is this (ALRS6 8157 JJ) model to old for this project?

### DISCUSSION THREAD

**Discussion by tccyp001** [aqualinkd/AqualinkD#161] [Q&A] [answered]
Amazing project! it is like 2-3 full time engineers whole year's job.

I have a indoor controller RS6 control panel, I thought AqualinkD should work. 
brought the hardware, installed, didn't work.
serial_logger doesn't show the Control Panel Model and version.

Opened the outside control box, and found the chipset model is  ALRS6 8157 JJ, 
Is this one too old for this project? (I brought the house last year, not sure when previous owner install the Aqualink)

if it is too old, does it mean if I buy a new RS8 combo panel and upgrade myself? Not super clear about how the pool system works.

**Comment by ballle98** [comment]
I suspect you do not have the wiring correct see https://github.com/sfeakes/AqualinkD/wiki#RS485 or the RS485 serial adapter you have is not good.  you should be getting commands back and forth on the serial port.  You hook it up in parallel on the same port that it uses to talk to the pumps.

BTW I have been using this RS485 adapter for 3 years flawless - https://smile.amazon.com/gp/product/B01LCFRR3E/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
I use CAT5e wire and ran a wire about 20 ft from the panel up a wall to inside.  That way my pi is indoors and I don't  have to worry about it.

**Reply by tccyp001** [reply]
Thanks for your suggestion, this adapter actually is better than the one I brought earlier (easy to connect in my case)

**Comment by sfeakes** [comment]
ALRS6 8157 JJ Will be fine. as above, either wiring / hardware adapter or OS serial problem.   Most people who have this issue it’s simply a case of swapping the data + & - wires around.  But keypad termination can also be an issue, way to check that is to disconnect the data + & - from the keypad.

**Comment by tccyp001** [comment]
Thanks for the quick response, I disconnected the keypad and connected black and yellow to this one according to the instruction.  https://www.amazon.com/gp/product/B01HI9CMT2/ref=ppx_yo_dt_b_asin_title_o05_s01?ie=UTF8&psc=1
thanks for the quick response, I am using https://www.amazon.com/gp/product/B01HI9CMT2/ref=ppx_yo_dt_b_asin_title_o05_s01?ie=UTF8&psc=1 
I double checked the wiring (even checked the outside box) looks good.
I do see some data in the serial port, but didn't give me the panel model, I manually configure the config file to RS-6 Combo, 
was able to run the aqualinkd, but on the ui it keep showing connecting.
here is the log for serial_logger, do you think it is normal?

pi@raspberrypi:~/software/AqualinkD $ sudo ./release/serial_logger /dev/ttyUSB0 -p 1000
AqualinkD serial_logger V1.4
Notice:  RS Serial: Logging serial information!
Please wait.|

Notice:  RS Serial: RS485 interface received 1000 packets in 99 seconds (~10.10 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   :
Notice:  RS Serial: Jandy Control Panel Version :
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0x30 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x31 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x32 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x33 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x48 is not used  <-- can use for Aqualinkd (RSSA ID)
Notice:  RS Serial: ID 0x49 is not used  <-- can use for Aqualinkd (RSSA ID)
Notice:  RS Serial: ID 0x08 is not used  <-- can use for Aqualinkd
Notice:  RS Serial: ID 0x09 is not used  <-- can use for Aqualinkd

**Reply by sfeakes** [reply]
Please ignore my last post, I also deleted it, I was reading it on a phone and missed the last 2 lines.  

Your output is normal, as long as you use those IDs in the config you should be good to go.  The fact it didn’t pick up the model is fine.  It’s the IDs that’s most important.

**Reply by tccyp001** [reply]
does that mean I only need change device_id=0x0a to device_id=0x08 line aqualinkd.conf? (or maybe I don't need change default 0x0a is good enough?)  
I think I tried before, but the UI keep saying "connecting" and when I click the button nothing happening on the pump etc.
Not sure whether there is way to debug the serial port.  Is there a easy way to for example just send some Hex through serial port to turn on/off something?


Also in the doc you mentioned "make sure all Pumps are on, and all keypads are awake " 
should I use keypad to turn the pump on and disconnect the serial cable and connect to raspberry and run serial_logger?

**Comment by sfeakes** [comment, marked as answer]
Yes you should change that ID.  These are the lines you should try in the config. (obviously find the appropriate place so you don't duplicate / override them)
```
panel_type = RS-6 Combo
device_id=0x09
rssa_device_id=0x48
extended_device_id=0x31
extended_device_id_programming = yes
```
The above is assuming you have a Combo unit (ie pool AND spa) modes.  Of not, change "RS-6 Combo" to "RS-6 Only"

First time you start, leave it running for a new minutes, let it connect and sort itself out, this can take ~2 mins.

NOTE. Most keypads are 0x08 by default, so the above config is assuming that.  What you can do once it all looks good / working, is plug the keypad back in, and run the serial logger again. The serial logger will tell you what ID the keypad is using, then obviously don't use the same ID as the keypad in the aqualinkd config.

**Reply by tccyp001** [reply]
working now! I think before I didn't uncomment device_id rssa_device_id part
I only set the panel_type, (I guess for that part the instruction is a little bit unclear)
but anyway thanks a lot for your help, great project. just sent you "cups of coffee" through paypal.  Enjoy your weekend!
